core/src/test/scala/unit/kafka/server/metadata/BrokerMetadataListenerTest.scala

```diff
@@ -18,28 +18,31 @@
 package kafka.server.metadata
 
 import java.util
+import java.util.concurrent.atomic.AtomicReference
 import java.util.{Collections, Optional}
 
 import org.apache.kafka.common.{Endpoint, Uuid}
-import org.apache.kafka.common.metadata.RegisterBrokerRecord
+import org.apache.kafka.common.metadata.{PartitionChangeRecord, PartitionRecord, RegisterBrokerRecord, TopicRecord}
 import org.apache.kafka.common.utils.Time
 import org.apache.kafka.image.{MetadataDelta, MetadataImage}
 import org.apache.kafka.metadata.{BrokerRegistration, RecordTestUtils, VersionRange}
 import org.apache.kafka.server.common.ApiMessageAndVersion
-import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.{assertEquals, assertTrue}
 import org.junit.jupiter.api.Test
 
 
 class BrokerMetadataListenerTest {
   @Test
   def testCreateAndClose(): Unit = {
-    val listener = new BrokerMetadataListener(0, Time.SYSTEM, None)
+    val listener = new BrokerMetadataListener(0, Time.SYSTEM, None, 1000000L,
+      (_, _, _, _) => throw new UnsupportedOperationException())
```

Comment:
nit: any harm using `MetadataSnapshotter` here and below?

Reply:
It starts a thread, which we would have to stop. Also I'd like to be sure that we didn't try to create a snapshot in this case (so, didn't see the UVE exception)